repo/repo.go

```diff
@@ -55,6 +55,9 @@ type Repo struct {
 
 	// Sum is the hash of the module to be verified after download.
 	Sum string
+
+	// File is a pointer to the file where the repository rule was defined.
+	File *rule.File
```

Comment:
It took me a long time to understand what `ListRepositories` and `FilterRulesByFile` were doing, and I think this is what was confusing me. I don't think this field belongs here. `Repo` should be describing a repository, not a declaration.

We do need a way for `ListRepositories` to return the parsed files and to indicate which repos were declared where. Maybe it could return a `map[*File][]string` where the strings are the repo names, and then `FilterRulesByFile` could do something with that?

Reply:
Sorry about that, I had a feeling this part may be confusing. `ListRepositories` new signature will look like this:
`func ListRepositories(workspace *rule.File) (repos []Repo, reposByFile map[*rule.File][]string, err error)`

Reply:
👍